### PR TITLE
chore: add UpdateFeature

### PR DESCRIPTION
**Pull request by kentakozuka**
part of https://github.com/bucketeer-io/bucketeer/issues/904

### CODE REVIEW

File: proto/event/domain/event.proto [hunk 1]
```diff
@@ -258,6 +259,11 @@ message FeatureCreatedEvent {
   repeated bucketeer.feature.Target targets = 12;
 }
 
+message FeatureUpdatedEvent {
+  string id = 1;
+  string data = 2;
```
Comment (kentakozuka) [inline proto/event/domain/event.proto:264]: `data` contains a JSON-formatted feature object. We can show diffs between the later and previous data by comparing this data.

Comment (kentakozuka) [reply proto/event/domain/event.proto:264]:
```suggestion
  string data = 2;
  string prev_data = 3;
```

File: proto/backend/service.proto [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  google.protobuf.FieldMask field_mask = 3;
+  feature.Feature feature = 4;
+}
+
+message UpdateFeatureResponse {}
+
+service PublicAPIService {
+  rpc GetFeature(GetFeatureRequest) returns (GetFeatureResponse) {
+    option (google.api.http) = {
+      get: "environment/{environment-namespace}/features/{id}"
```
Comment (cre8ivejp) [inline proto/backend/service.proto:46]:
```suggestion
      get: "environment/{environment-id}/features/{id}"
```

File: proto/backend/service.proto [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    };
+  }
+  rpc UpdateFeature(UpdateFeatureRequest) returns (UpdateFeatureResponse) {
+    option (google.api.http) = {
+      patch: "environment/{environment-namespace}/features/{id}"
```
Comment (cre8ivejp) [inline proto/backend/service.proto:51]:
```suggestion
      patch: "environment/{environment-id}/features/{id}"
```

File: proto/backend/service.proto [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  string id = 2;
+}
+
+message GetFeatureResponse {
+  feature.Feature feature = 1;
+}
+
+message UpdateFeatureRequest {
+  string comment = 1;
+  string environment_namespace = 2;
```
Comment (cre8ivejp) [inline proto/backend/service.proto:36]:
```suggestion
  string environment_id = 2;
```

File: proto/backend/service.proto [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+package bucketeer.backend;
+option go_package = "github.com/bucketeer-io/bucketeer/proto/backend";
+
+import "google/api/annotations.proto";
+import "google/protobuf/field_mask.proto";
+
+import "proto/feature/feature.proto";
+
+message GetFeatureRequest {
+  string environment_namespace = 1;
```
Comment (cre8ivejp) [inline proto/backend/service.proto:26]:
```suggestion
  string environment_id = 1;
```

File: proto/feature/service.proto [hunk 6]
```diff
@@ -97,6 +98,15 @@ message CreateFeatureRequest {
 
 message CreateFeatureResponse {}
 
+message UpdateFeatureRequest {
+  string comment = 1;
+  string environment_namespace = 2;
```
Comment (cre8ivejp) [inline proto/feature/service.proto:103]:
```suggestion
  string environment_id = 2;
```

File: proto/backend/service.proto [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+message GetFeatureResponse {
+  feature.Feature feature = 1;
+}
+
+message UpdateFeatureRequest {
+  string comment = 1;
+  string environment_namespace = 2;
+  google.protobuf.FieldMask field_mask = 3;
```
Comment (cre8ivejp) [inline proto/backend/service.proto:37]: Where are you going to use this?

File: pkg/backend/api/grpcapi.go [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func (s *BackendService) GetFeature(
+	ctx context.Context,
+	req *backendproto.GetFeatureRequest,
+) (*backendproto.GetFeatureResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "method not implemented")
+}
+
+func (s *BackendService) UpdateFeatureFeature(
```
Comment (cre8ivejp) [inline pkg/backend/api/grpcapi.go:66]:
```suggestion
func (s *BackendService) UpdateFeature(
```

Comment (cre8ivejp) [inline pkg/backend/api/grpcapi.go:66]: What is the difference between this API and this [UpdateFeature](https://github.com/bucketeer-io/bucketeer/pull/1001/files#diff-2d0a2c8153246509abb80eca89b6ca25f95f7ea8e75640dae22408489b8b0eafR659-R665)

File: pkg/backend/api/grpcapi.go [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+) *BackendService {
+	dopts := &options{
+		logger: zap.NewNop(),
+	}
+	for _, opt := range opts {
+		opt(dopts)
+	}
+	return &BackendService{
+		opts:   dopts,
+		logger: dopts.logger.Named("api"),
```
Comment (cre8ivejp) [inline pkg/backend/api/grpcapi.go:55]:
```suggestion
		logger: dopts.logger.Named("backend/api"),
```

File: proto/backend/service.proto (hunk 7 above)
Comment (kentakozuka) [reply proto/backend/service.proto:37]: field mask is for partial update
https://protobuf.dev/programming-guides/api/#use-update-field-mask

File: pkg/backend/api/grpcapi.go (hunk 8 above)
Comment (kentakozuka) [reply pkg/backend/api/grpcapi.go]: This method is for public API and this [UpdateFeature](https://github.com/bucketeer-io/bucketeer/pull/1001/files#diff-2d0a2c8153246509abb80eca89b6ca25f95f7ea8e75640dae22408489b8b0eafR659-R665) is for UI.

**Comment by kentakozuka** [comment]
@cre8ivejp Fixed 👍

File: pkg/backend/api/grpcapi.go (hunk 8 above)
Comment (cre8ivejp) [reply pkg/backend/api/grpcapi.go]: Will the APIs in the backend pkg be a proxy layer for the authentication only?

File: proto/backend/service.proto (hunk 7 above)
Comment (kentakozuka) [reply proto/backend/service.proto]: @cre8ivejp I changed my mind. I removed the field mask.
Usthe ing field mbringsring complextoy of UI logic. I want to start by being straightforward.
WDYT?

File: pkg/backend/api/grpcapi.go (hunk 8 above)
Comment (kentakozuka) [reply pkg/backend/api/grpcapi.go]: Yes. Tha's what I plan so far :)

File: proto/backend/service.proto (hunk 7 above)
Comment (cre8ivejp) [reply proto/backend/service.proto]: Agree. Let's try to use it later if needed.

**Comment by kentakozuka** [comment]
@cre8ivejp Fixed 👍

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**